iset.mm

```diff
@@ -103467,68 +103480,122 @@ Cauchy sequence (for example, fixing a rate of convergence as in
        partial sums, if it exists.  This means that conditionally convergent
        sums can be evaluated meaningfully.  For finite sums, we are explicitly
        order-independent, by picking any bijection to a 1-based finite sequence
-       and summing in the induced order.  These two methods of summation
-       produce the same result on their common region of definition (i.e.
-       finite subsets of the upper integers).  Examples:
-       ` sum_ k e. { 1 , 2 , 4 } ` ` k ` means ` 1 + 2 + 4 = 7 ` , and
-       ` sum_ k e. NN ` ` ( 1 / ( 2 ^ k ) ) = 1 ` means 1/2 + 1/4 + 1/8 + ... =
-       1.  (Contributed by NM, 11-Dec-2005.)  (Revised by Mario Carneiro,
-       13-Jun-2019.) $)
-    df-sum $a |- sum_ k e. A B = ( iota x ( E. m e. ZZ ( A C_ ( ZZ>= ` m ) /\
-     seq m ( + , ( n e. ZZ |-> if ( n e. A , [_ n / k ]_ B , 0 ) ) , CC )
-       ~~> x ) \/
-     E. m e. NN E. f ( f : ( 1 ... m ) -1-1-onto-> A /\
-       x = ( seq 1 ( + , ( n e. NN |-> [_ ( f ` n ) / k ]_ B ) , CC ) ` m )
-     ) ) ) $.
-
+       and summing in the induced order.  In both cases we have an ` if `
+       expression so that we only need ` B ` to be defined where ` k e. A ` .
+       These two methods of summation produce the same result on their common
+       region of definition (i.e. finite subsets of the upper integers).
+       Examples: ` sum_ k e. { 1 , 2 , 4 } ` ` k ` means ` 1 + 2 + 4 = 7 ` ,
+       and ` sum_ k e. NN ` ` ( 1 / ( 2 ^ k ) ) = 1 ` means 1/2 + 1/4 + 1/8
+       + ... = 1.  (Contributed by Jim Kingdon, 15-Feb-2022.) $)
+    df-isum $a |- sum_ k e. A B = ( iota x ( E. m e. ZZ ( A C_ ( ZZ>= ` m ) /\
+      A. j e. ( ZZ>= ` m ) DECID j e. A /\
```

Comment:
Maybe add something like (if I understood correctly): In the inifinite case, we also require that the indexing set be a decidable subset of an upperset of integers (that is, membership of integers in it is decidable).

Reply:
I've added this wording.